references/cli/up.go

```diff
@@ -68,6 +68,12 @@ func NewUpCommand(c common2.Args, order string, ioStream cmdutil.IOStreams) *cob
 			if err != nil {
 				return errors.Wrap(err, "File format is illegal, only support vela appfile format or OAM Application object yaml")
 			}
+
+			// Override namespace if namespace flag is set. We should check if namespace is `default` or not
+			// since GetFlagNamespaceOrEnv returns default namespace when failed to get current env.
+			if namespace != "" && namespace != types.DefaultAppNamespace {
```

Comment:
I found an edge case that `GetFlagNamespaceOrEnv` returns `default` namespace if failed to call `GetCurrentEnv`. So I have to add `default` namespace check.

Could you double-check for this?

https://github.com/oam-dev/kubevela/blob/947455adb998426d56139f128b3a9cd7100e1d35/references/cli/env.go#L243-L249